pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -54,7 +55,45 @@ func NewController() *Controller {
 func (c *Controller) AddRegistry(registry Registry) {
 	c.storeLock.Lock()
 	defer c.storeLock.Unlock()
-	c.registries = append(c.registries, registry)
+	registries := c.registries
```

Comment:
Why introduce the temp variable `registries`? I think the previous logic is good enough as we already have lock before append, am I missing anything here?

Same question as follows where using the temp variable of `registries`.

Reply:
@gyliu513 the idea is to protect from inconsistency any go routine which might be reading the list at the time of adding/deleting. With this swap, we make sure that controller's registries points to the updated version. Does it make sense to you?

Reply:
well, I don't think this is necessary. Basically, the two pointers (registries, c.registries) will be identical. The append may or may not return a different pointer, so it's not something dependable.